posthog/hogql/database/schema/persons.py

```diff
@@ -162,11 +177,56 @@ def to_printed_hogql(self):
         return "raw_persons"
 
 
+# Persons is a lazy table that allows you to insert a where statement inside of the person subselect
+# It pulls any "persons.id in ()" statement inside of the argmax subselect
+# This is useful when executing a query for a large team.
+# It could be useful to set "use_query_cache" if the query is repeated. See `actors_query_runner.py` for an example
 class PersonsTable(LazyTable):
     fields: dict[str, FieldOrTable] = PERSONS_FIELDS
+    filter: Optional[Expr] = None
+
+    @staticmethod
+    def _is_promotable_expr(expr, alias: Optional[str] = None):
+        return (
+            isinstance(expr, CompareOperation)
+            and expr.op == CompareOperationOp.In
+            and isinstance(expr.left, Field)
+            and expr.left.chain == [alias or "persons", "id"]
+        )
+
+    @staticmethod
+    def partition_exprs(exprs, alias: Optional[str] = None):
+        not_promotable = []
+        promotable = []
+        for expr in exprs:
+            if PersonsTable._is_promotable_expr(expr, alias):
+                # Erase "persons" from the chain before bringing inside
+                expr.left = Field(chain=["id"])
+                promotable.append(expr)
+            else:
+                not_promotable.append(expr)
+
+        return promotable, not_promotable
+
+    # If the join has a clause we can bring inside the subselect, create a new table that represents that
+    def create_new_table_with_filter(self, join: JoinExpr) -> Self:
+        if join.constraint is not None and isinstance(join.constraint.expr, And):
+            exprs = cast(And, join.constraint.expr).exprs
+            promotable, not_promotable = PersonsTable.partition_exprs(exprs, join.alias)
+            join.constraint.expr.exprs = not_promotable
+            p = PersonsTable()
```

Comment:
Why a blank slate `PersonsTable` instead of in place or `.model_copy()`?

Reply:
Model copy makes sense.

In place doesn't work because different persons joins in the same query can have different filters.